### ISSUE TITLE
[bug] TypeError: sequence item 0: expected str instance, NoneType found

### ISSUE THREAD

**Issue by DoDoENT** [conan-io/conan#16086]
### Describe the bug

When using profile with custom architecture that represents an apple universal binary, conan fails with:

```
ERROR: log_and_timer/1.1.3@nenad/testing: Error in generate() method, line 56
        self.mb_generate_with_cmake_args(cmake_args)
while calling 'mb_generate_with_cmake_args', line 65
        tc.generate()
        TypeError: sequence item 0: expected str instance, NoneType found
```

`mb_genererate_with_cmake_args` is our function that sets up the `CMakeToolchain` with variables and then calls the `generate` function on it.

The error was introduced in conan v2.2.0, possibly with [official support for universal binaries](https://github.com/conan-io/conan/pull/15775/files).

Downgrading to v2.1.0 fixes the issue.

I also tried setting `arch = armv8|x86_64` in my profile instead of the custom `universal`, however this generates a broken xcode project which fails to build (the error being `make: *** No rule to make target `/Users/dodo/.conan2/p/b/log_a9bf66684d376f/b/build/generators/Boost-release-armv8', needed by `/Users/dodo/.conan2/p/b/log_a9bf66684d376f/b/build/CMakeFiles/cmake.check_cache'.  Stop.`). If I tried using `ninja` instead of `xcode`, then it fails with

```
CMake Error:
  Running

   '/opt/homebrew/bin/ninja' '-C' '/Users/dodo/.conan2/p/b/gtest5524cfbfaad17/b/build/Release' '-t' 'recompact'

  failed with:

   ninja: error: build.ninja:527: expected newline, got '|'
```

### How to reproduce it

- add a custom architecture to your `settings_user.yml`
- use it in mac or iOS profile
- try building the conan package using that profile with CMakeToolchain

**Comment by DoDoENT** [comment]
For reference, here are the profiles I use (it may help you with triaging):

```======== Input profiles ========
Profile host:
[settings]
arch=universal
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15.0.0
os=iOS
os.sdk=iphonesimulator
os.version=11.0
[tool_requires]
*: mb_foundation/1.0.1@microblink/main
[conf]
tools.cmake.cmake_layout:build_folder_vars=['settings.os', 'settings.os.sdk', 'settings.arch', 'settings.compiler', 'settings.compiler.version', 'settings.compiler.sanitizers', 'settings.compiler.link_time_optimization']
tools.cmake.cmaketoolchain:generator=Xcode
tools.cmake.cmaketoolchain:system_name=iOS
tools.files.download:retry=25
tools.files.download:retry_wait=2
user.microblink.cmaketoolchain:cache_variables={'MB_IOS_SDK': 'simulator'}

Profile build:
[settings]
arch=armv8
build_type=Release
compiler=apple-clang
compiler.libcxx=libc++
compiler.version=15.0.0
os=Macos
[tool_requires]
*: mb_foundation/1.0.1@microblink/main
[conf]
tools.cmake.cmake_layout:build_folder_vars=['settings.os', 'settings.os.sdk', 'settings.arch', 'settings.compiler', 'settings.compiler.version', 'settings.compiler.sanitizers', 'settings.compiler.link_time_optimization']
tools.cmake.cmaketoolchain:generator=Ninja
tools.files.download:retry=25
tools.files.download:retry_wait=2
```

**Comment by czoido** [comment]
Hi @DoDoENT,

Thanks a lot for reporting, I have been able to reproduce the issue and opened a [PR to fix that](https://github.com/conan-io/conan/pull/16090).
This will be released in Conan 2.2.3 soon

**Comment by czoido** [comment]
@DoDoENT closing this after https://github.com/conan-io/conan/pull/16090 included in version 2.2.3 that has just been released: https://github.com/conan-io/conan/releases/tag/2.2.3
Thanks a lot for reporting.

**Comment by DoDoENT** [comment]
Thank you!
